Add Docker configuration

senqpaupa opened the pull request:
Creating three Docker containers. Each container is responsible for a separate functionality, making it easier to deploy and manage the application.

Changes:
1. Nginx Container:
   - Configured to handle HTTP requests.
   - Nginx configuration has been added to docker/nginx/conf.d.

2. PHP Container:
   - Handles the execution of PHP scripts.
   - Configured to work with Nginx via FastCGI.

3. Database Container:
   - Added a container for managing application data.
   - Environment variables have been configured for database connection.

Strucure
├── nginx/
│ ├── Dockerfile
│ └── conf/
│ └── default.conf
├── php/
│ └── Dockerfile
├── postgres/
│ ├── Dockerfile
│ └── init.sql
└── app/
└── public/
└── index.php

praisethedeviI commented on docker/php/docker-compose.yml:
```diff
@@ -0,0 +1,43 @@
+version: '3.8'
```
этот файл предлагаю вынести в корень проекта, а не держать где-то далеко внутри docker/php

senqpaupa replied:
Исправил

praisethedeviI commented on nginx/Dockerfile:
```diff
@@ -0,0 +1,17 @@
+FROM nginx:alpine
```
лишний файл, уже есть же docker/nginx/Dockerfile

senqpaupa replied:
Удалил

praisethedeviI commented on nginx/conf/default.conf:
```diff
@@ -0,0 +1,21 @@
+server {
```
то же самое

senqpaupa replied:
Удалил

praisethedeviI commented on php/Dockerfile:
```diff
@@ -0,0 +1,15 @@
+FROM php:8.2-fpm
```
Лишнее, есть docker/php/Dockerfile уже

senqpaupa replied:
Удалил

praisethedeviI commented on postgres/Dockerfile:
```diff
@@ -0,0 +1,2 @@
+FROM postgres:15-alpine
```
файл предлагаю перенести в директорию docker/postgres. Пусть все докерское лежит там, а docker-compose.yml из под корня проекта в это все смотрит

senqpaupa replied:
Изменил структуру проекта

praisethedeviI commented on docker/php/docker-compose.yml:
```diff
@@ -0,0 +1,43 @@
+    image: postgres:15-alpine
+    ports:
+      - "5432:5432"
+    environment:
+      POSTGRES_DB: myapp
+      POSTGRES_USER: user
+      POSTGRES_PASSWORD: password
```
Это должно настраиваться через .env файл, это приватные данные

senqpaupa replied:
Создал env.example и добавил в git.ignore env, но во втором PR.

praisethedeviI replied:
Не вижу изменений по .env. Там должны быть переменные для базы данных

> Создал env.example и добавил в git.ignore env, но во втором PR.

В .gitignore должны быть .env.local и .env.*.local, не .env. Если .gitgnore еще не настроен правильно, то это надо сделать в этом PR, не следующем

praisethedeviI commented:
Название коммитов и PR начинаем с глаголов в инфинитиве

praisethedeviI commented:
В PR не попал коммит с развертыванием проекта

feip-krk commented on docker-compose.yml:
```diff
@@ -0,0 +1,43 @@
+    environment:
+      POSTGRES_DB: myapp
+      POSTGRES_USER: user
+      POSTGRES_PASSWORD: password
```
все еще не вынес это в env файл
например так:
```
      - POSTGRES_DB=${DB_DATABASE}
      - POSTGRES_USER=${DB_USERNAME}
      - POSTGRES_PASSWORD=${DB_PASSWORD}
```

у тебя должно быть два енв, один локальный (скрытый) .env.local и один публичный .env

senqpaupa replied:
Исправил

praisethedeviI replied:
тут ничего не поменялось

feip-krk commented:
Объедини все коммиты в один в этом пулл-ревесте и напиши им хороший текст, например, как в заголовке пулл-реквеста

senqaa committed:
docker nginx php

Создал 2 зависимых контейнера nginx и php

nginx container

независимый контейнер nginx

php container

Независимый php контейнер

Postgres container

Независимый postgres контейнер

Bug fix

Buf fix

praisethedeviI commented on docker/postgres/Dockerfile:
```diff
@@ -0,0 +1,8 @@
+FROM postgres:15-alpine
+
+
+ENV POSTGRES_DB=${POSTGRES_DB}
+ENV POSTGRES_USER=${POSTGRES_USER}
+ENV POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
```
Не тут, в docker-compose.yml надо

senqaa committed:
fix private data

senqaa committed:
add .env.example

senqaa committed:
update .env

senqaa committed:
.env fix

feip-krk commented:
.env.example лишний
